### PR TITLE
filesystem creation for ceph

### PR DESCRIPTION
**Pull request by shreekarSS**
Added template for filesystem, updated rook resource creation for filesystem.yaml

### CODE REVIEW

File: tests/test_ocs_basic_install.py [hunk 1]
```diff
@@ -160,7 +160,7 @@ def run(**kwargs):
     create_oc_resource(
         "prometheus-rules.yaml", rook_data, cluster_path, _templating
     )
-
+    create_oc_resource('filesystem.yaml', rook_data, cluster_path, _templating)
```
Comment (vasukulkarni) [inline tests/test_ocs_basic_install.py]: Lets move this in Rook Class as `def create_filesystem()`,  I think once I see the admin guide, we can revisit how to handle deployment with required services.

**Comment by vasukulkarni** [comment]
@shreekarSS are you working on this one along with a test?

**Comment by shylesh** [comment]
Approximately it should look like , rookcluster.create_fs(param1, param2 ...). 
create_fs function should consider creating the fs resource yaml by putting params into template and invoke creation through create call to openshift. Also this function should do verification of fs creation in the cluster.

roughly:

class RookCluster():
    def create_fs(param1, param2...):
          service_data = "
          api_ver: <v>
          expand yaml by replacing params
          "
          api_client.create(body=body, namespace=self.namespace)

**Comment by RazTamir** [comment]
Why rook cluster and not OCS?

**Comment by shylesh** [comment]
> Why rook cluster and not OCS?

Because a ceph filesystem OR RBD image closely tied with a ceph cluster.   It would be inherently meaningful to access something like cluster.fs OR cluster.RBD etc instead of treating them as independent units.  In case if you create a OCS class I feel ceph cluster will be a composition of OCS class because OCS contains a ceph cluster along with other non-ceph entities  and a ceph cluster contains filesystem OR RBD.

My 2 cents.

File: templates/ocs-deployment/filesystem.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#################################################################################################################
+# Create a filesystem with settings with replication enabled for a production environment.
+# A minimum of 3 OSDs on different nodes are required in this example.
+#  kubectl create -f filesystem.yaml
+#################################################################################################################
```
Comment (mbukatov) [inline templates/ocs-deployment/filesystem.yaml]: If you have used [this file](https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/filesystem.yaml) as a base here, please make it clear in the comment.

Also if that is the case, I would even suggest to add the file in it's original form in one commit, and then changing it for the purposes of ocs qe work.

**Comment by vasukulkarni** [comment]
> Why rook cluster and not OCS?

the current class is named Rook :)

File: templates/ocs-deployment/filesystem.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      size: 3
+  # The list of data pool specs. Can use replication or erasure coding.
+  dataPools:
+    - failureDomain: host
+      replicated:
+        size: 3
+  # The metadata service (mds) configuration
+  metadataServer:
+    # The number of active MDS instances
+    activeCount: 1
```
Comment (ramkiperiy) [inline templates/ocs-deployment/filesystem.yaml]: Please make this as param, where default 1 MDS count, or the user can specify the count.

File: templates/ocs-deployment/filesystem.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+spec:
+  # The metadata pool spec. Must use replication.
+  metadataPool:
+    replicated:
+      size: 3
+  # The list of data pool specs. Can use replication or erasure coding.
+  dataPools:
+    - failureDomain: host
+      replicated:
+        size: 3
```
Comment (ramkiperiy) [inline templates/ocs-deployment/filesystem.yaml]: change this as a param "replicasize"

Comment (shreekarSS) [reply templates/ocs-deployment/filesystem.yaml]: yes Ram, I have made func create_fs() with all  configurable params needed

File: templates/ocs-deployment/filesystem.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
+#################################################################################################################
+# Create a filesystem with settings with replication enabled for a production environment.
+# A minimum of 3 OSDs on different nodes are required in this example.
+#  kubectl create -f filesystem.yaml
+#################################################################################################################
+
+apiVersion: ceph.rook.io/v1
+kind: CephFilesystem
+metadata:
+  name: myfs
```
Comment (petr-balogh) [inline templates/ocs-deployment/filesystem.yaml]: Shouldn't be the name also parametrized?

File: templates/ocs-deployment/filesystem.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  namespace: {{ cluster_namespace | default('openshift-storage') }}
+spec:
+  # The metadata pool spec. Must use replication.
+  metadataPool:
+    replicated:
+      size: 3
```
Comment (petr-balogh) [inline templates/ocs-deployment/filesystem.yaml]: Will there be the case when we want to have different replica size? If so, make it parametrized with default value 3.

File: templates/ocs-deployment/filesystem.yaml (hunk 3 above)
Comment (petr-balogh) [reply templates/ocs-deployment/filesystem.yaml]: +1

File: templates/ocs-deployment/filesystem.yaml (hunk 4 above)
Comment (petr-balogh) [reply templates/ocs-deployment/filesystem.yaml]: +1 on parameterized here as well

**Comment by RazTamir** [comment]
@shreekarSS , is this patch still needed?

**Pull request closed without merging**